### ISSUE TITLE
Column sorting does not sort correctly after scrolling horizontally

### ISSUE THREAD

**Issue by JellyRollMorton** [handsontable/handsontable#542]
After I scroll horizontally to the right, so that the first column is no longer visible, then columns are not sorted in the correct order.  Here is an example:

http://jsfiddle.net/r2b4S/

In the example, if the left column shown is Column C, then clicking the header for Column C actually appears to sort by Column A data.  Clicking the Column E header appears to sort by Column C data.  It seems like the sorting is not taking into account the visible offset of the columns after scrolling.

```javascript
        if ($target.is('.columnSorting')) {
          var col = $target.closest('th').index();
```

to:

```javascript
        if ($target.is('.columnSorting')) {
          var col = $target.closest('th').index() + instance.colOffset();   
```

I'm not sure if this is the proper fix in this case, but it appears to be working well for me.

**Comment by agottem** [comment]
This seems to be broken again in the latest version

**Comment by krzysztofspilka** [comment]
@agottem thanks for reporting, reopened just now

**Comment by jansiegel** [comment]
Should work fine after releasing `0.21.0`, with the fix from #3014
